#VRML_SIM R2019a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.3172443932135793 0.9365667617215259 0.1489922676282317 0.8869141482249077
  position 2.4012963532779317 0.9743204444645783 1.7316526517979498
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
Robot {
  translation 0.1 0.07999999999999997 0.1
  children [
    DEF GR_BODY Group {
      children [
        DEF ROBOT_BODY Shape {
          appearance PBRAppearance {
            baseColor 0.9372549019607843 0.1607843137254902 0.1607843137254902
          }
          geometry DEF G_BODY Box {
            size 0.25 0.07 0.1
          }
        }
        Transform {
          translation 0.11 -0.05 0
          children [
            Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Sphere {
                radius 0.015
              }
            }
          ]
        }
      ]
    }
    DEF WHEEL1_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor -0.05 0 0.06
      }
      device [
        PositionSensor {
          name "encoder1"
        }
        RotationalMotor {
          name "motor_right"
          maxTorque 30
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation -0.05 0 0.06
        rotation -1 0 0 1.57
        children [
          DEF SH_WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry DEF G_WHEEL Cylinder {
              height 0.02
              radius 0.075
            }
          }
        ]
        boundingObject USE G_WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor -0.05 0 -0.06
      }
      device [
        PositionSensor {
          name "encoder2"
        }
        RotationalMotor {
          name "motor_left"
          maxTorque 30
        }
      ]
      endPoint Solid {
        translation -0.05 0 -0.06
        rotation -1 0 0 1.57
        children [
          DEF WH_SHAPE2 Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.075
            }
          }
        ]
        name "solid(1)"
        boundingObject USE WH_SHAPE2
        physics Physics {
        }
      }
    }
    DistanceSensor {
      translation 0.125 0 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.1803921568627451 0.20392156862745098 0.21176470588235294
          }
          geometry Box {
            size 0.02 0.01 0.01
          }
        }
      ]
      name "distance_sensor"
      lookupTable [
        0 0 0
        0.02 0 0
        4 1023 0
      ]
      resolution 1
    }
  ]
  boundingObject USE GR_BODY
  physics Physics {
  }
  controller "u2_wp_Zamudio_Shadeen"
}
Solid {
  translation 0.8700000000000006 0 0.16
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.9882352941176471 0.9137254901960784 0.30980392156862746
      }
      geometry Box {
        size 0.2 0.5 0.3
      }
    }
  ]
}
